gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsInfoApiMonitor.java

```diff
@@ -47,18 +63,42 @@ public ClusterStats monitor(ProxyBackendConfiguration backend)
     }
 
     private boolean isReadyStatus(String baseUrl)
+    {
+        return isReadyStatus(baseUrl, retries);
+    }
+
+    private boolean isReadyStatus(String baseUrl, int retriesRemaining)
     {
         Request request = prepareGet()
                 .setUri(uriBuilderFrom(URI.create(baseUrl)).appendPath("/v1/info").build())
                 .build();
-
         try {
             ServerInfo serverInfo = client.execute(request, SERVER_INFO_JSON_RESPONSE_HANDLER);
             return !serverInfo.isStarting();
         }
+        catch (UnexpectedResponseException e) {
+            if (shouldRetry(e.getStatusCode())) {
+                if (retriesRemaining > 0) {
+                    return isReadyStatus(baseUrl, retriesRemaining - 1);
```

Comment:
I'd log something to inform that the retries are happening. Maybe something like `log.warn("received an retriable error, retrying...")